code/datums/sound_player.dm

```diff
@@ -224,6 +224,9 @@ datum/sound_token/proc/Mute()
 	if(listener in listeners)
 		return
 
+	if(!istype(listener, /mob))
```

Comment:
this safety check seems unnecessary, and it does overlap with the virtualmob check above, consider reordering things here